File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/Task.kt
package com.example.diyhomeautomation.models

import java.time.LocalTime

/**
 * Data class representing tasks in DIY home automation.
 *
 * @property id The unique identifier for the task.
 * @property name The name of the task.
 * @property state The state of the task.
 * @property initHour The initial time of the task.
 * @property endHour The ending time of the task.
 * @property daysOfWeek The days of the week when the task is scheduled.
 * @property taskDevices List of devices associated with the task (nullable).
 *
 * @constructor Creates an instance of Task with the provided properties.
 */
data class Task(
    var id: Int,
    var name: String,
    var state: Boolean,
    var initHour: LocalTime,
    var endHour: LocalTime,
    var daysOfWeek: String,
    var taskDevices: List<TaskDevice>?
)

File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/Task.cs
using DIYHomeAutomationAPI.Models.Enums;

namespace DIYHomeAutomationAPI.Models
{
    /// <summary>
    /// This class represents a Task
    /// </summary>
    public class Task
    {
        /// <summary>
        /// Task's Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Task's Name
        /// </summary>
        public string Name { get; set; } = null!;

        /// <summary>
        /// Task's State
        /// </summary>
        public bool State { get; set; }

        /// <summary>
        /// Task's InitHour
        /// </summary>
        public TimeSpan InitHour { get; set; }

        /// <summary>
        /// Task's EndHour
        /// </summary>
        public TimeSpan EndHour { get; set; }

        /// <summary>
        /// Task's Days of Weeks
        /// </summary>
        public string DaysWeek { get; set; } = null!;

        /// <summary>
        /// List of Task Devices
        /// </summary>
        public List<TaskDevice>? TaskDevices { get; set; }
    }
}
